File: src/chain-observers/classifiers/operation-classifier.ts
import type { operation } from "@hiveio/wax";
import { ITransactionData } from "./block-classifier";
import { CollectorClassifierBase } from "./collector-classifier-base";

export interface IOperationTransactionPair<OpType = operation> {
  operation: OpType;
  transaction: ITransactionData;
}

export interface IOperationBaseData {
  operations: Iterable<IOperationTransactionPair>;
}

export interface IOperationData extends IOperationBaseData {
  operationsPerType: {
    [key in keyof operation]: Iterable<IOperationTransactionPair<Exclude<operation[key], undefined>>>;
  };
}

export class OperationClassifier extends CollectorClassifierBase<{}, IOperationData> {}

File: python/workerbee/chain_observers/classifiers/operation_classifier.py
from __future__ import annotations

from ..classifier_results import OperationClassifierData
from .collector_classifier_base import CollectorClassifierBase


class OperationClassifier(CollectorClassifierBase[OperationClassifierData]):
    """IOperationData:
    operations: list[IOperationTransactionPair]
    operations_per_type: dict[str, list[IOperationTransactionPair]]
    """

    pass
